daffodil-core/src/main/scala/org/apache/daffodil/grammar/AlignedMixin.scala

```diff
@@ -260,6 +285,7 @@ trait AlignedMixin extends GrammarMixin { self: Term =>
           case LengthKind.Prefixed => LengthMultipleOf(1) // NYI
         }
       }
+      case mg: ModelGroup => LengthMultipleOf(1)
```

Comment:
So, this code path is actually getting hit by a negative test that is trying to trigger an SDE.  The test has an unordered sequence containing a choice, which is not allowed.  So what I think is happening is that the we are attempting to get the elementSpecifiedLengthApprox of this choice.
I think the proper fix for this is to use the Assert.usageError for now, and handle this case in the SequenceGrammarMixin so that it doesn't even try to make a SequenceChild from the ModelGroup in an unordered sequence.